File: app/src/main/java/me/gabeg/sicksends/trad/SsTradRepository.java
package me.gabeg.sicksends.trad;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemDatabase;
import me.gabeg.sicksends.problem.SsProblemRepository;

/**
 * Repository for trad problems.
 */
public class SsTradRepository
	extends SsProblemRepository<SsTradDao, SsTrad>
{

	/**
	 */
	public SsTradRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.tradDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/trad/SsTradRepository.kt
package me.gabeg.sicksends.trad

import me.gabeg.sicksends.problem.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for trad problems.
 */
class SsTradRepository @Inject constructor(dao : SsTradDao)
	: SsGenericProblemRepository<SsTradProblem>(dao)
